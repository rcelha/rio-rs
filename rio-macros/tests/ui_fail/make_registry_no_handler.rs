use rio_rs::prelude::*;
use serde::{Deserialize, Serialize};

use rio_macros::make_registry;

#[derive(Default, Debug, WithId, TypeName)]
struct TestService {
    id: String,
}

impl ServiceObjectStateLoad for TestService {}
impl ServiceObject for TestService {}

#[derive(TypeName, Message, Debug, Deserialize, Serialize)]
pub struct Ping {
    pub ping_id: String,
}

#[derive(TypeName, Message, Debug, Deserialize, Serialize)]
pub struct Pong {
    pub ping_id: String,
}

make_registry! {
    TestService: [
        Ping => (Pong, NoopError),
    ]
}

fn main() {
    let _registry = server::registry();
}
