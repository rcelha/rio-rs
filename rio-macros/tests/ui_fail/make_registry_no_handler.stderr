error[E0277]: the trait bound `TestService: Handler<Ping>` is not satisfied
  --> tests/ui_fail/make_registry_no_handler.rs:24:1
   |
24 | / make_registry! {
25 | |     TestService: [
26 | |         Ping => (Pong, NoopError),
27 | |     ]
28 | | }
   | |_^ the trait `Handler<Ping>` is not implemented for `TestService`
   |
note: required by a bound in `Registry::add_handler`
  --> $WORKSPACE/rio-rs/src/registry/mod.rs
   |
   |     pub fn add_handler<T, M>(&mut self)
   |            ----------- required by a bound in this associated function
   |     where
   |         T: 'static + Handler<M> + IdentifiableType + Send + Sync,
   |                      ^^^^^^^^^^ required by this bound in `Registry::add_handler`
   = note: this error originates in the macro `make_registry` (in Nightly builds, run with -Z macro-backtrace for more info)
